pkg/asset/agent/iso.go

```diff
@@ -0,0 +1,92 @@
+package agent
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+	"os"
+
+	"github.com/openshift/assisted-image-service/pkg/isoeditor"
+	"github.com/openshift/installer/pkg/agent/isosource"
+	"github.com/openshift/installer/pkg/asset"
+	"github.com/pkg/errors"
+)
+
+const (
+	agentISOFilename = "output/agent-installer.iso"
+)
+
+// ISO is an asset that generates the bootable image used to install clusters.
+type ISO struct {
+	File *asset.File
+}
+
+var _ asset.WritableAsset = (*ISO)(nil)
+
+// Dependencies returns the assets on which the Bootstrap asset depends.
+func (a *ISO) Dependencies() []asset.Asset {
+	return []asset.Asset{
+		&Ignition{},
+	}
+}
+
+// Generate generates the image file for to ISO asset.
+func (a *ISO) Generate(dependencies asset.Parents) error {
+	ignition := &Ignition{}
+	dependencies.Get(ignition)
+
+	// TODO: Use BaseIso asset once it is available
+	baseImage, err := isosource.EnsureIso()
+	if err != nil {
+		return err
+	}
+
+	ignitionByte, err := json.Marshal(ignition.Config)
+	if err != nil {
+		return err
+	}
+
+	ignitionContent := &isoeditor.IgnitionContent{Config: ignitionByte}
+	custom, err := isoeditor.NewRHCOSStreamReader(baseImage, ignitionContent, nil)
+	if err != nil {
+		return err
+	}
+	defer custom.Close()
+
+	output, err := os.Create(agentISOFilename)
```

Comment:
Hmm, I hadn't thought about the fact that the ISO is absolutely massive and there is no way we can save it in memory and let the asset management code write it to disk for us.